### PR TITLE
Add setting to hide mature content from ContentDB

### PR DESCRIPTION
**Pull request by rubenwardy**
EDIT by paramat: See #8143

### CODE REVIEW

**Comment by stujones11** [comment]
This seems like a reasonable solution, however, I wonder if it could be extended further into an actual rating system, so for example, we could _only_ show PEGI 3 rated content on the android version. This way mods would have to be specifically deemed suitable for a younger audience.

**Comment by paramat** [comment]
rubenwardy are you also going to sort through all content and add the rating to the mature content, before 5.0.0?

**Comment by rubenwardy** [comment]
I've removed the content for now, I will do that some time after

**Comment by paramat** [comment]
Ok.

I'm still sceptical that adding a user-set filter to CDB allows any content without raising the rating, do you have a link to the evidence for this? It seems to me a young user can (and often will) set the filter how they want and access mature content.

So it seems to me that the setting might have to be hardcoded for Android, maybe this should be done for now until we know for sure?

**Comment by rubenwardy** [comment]
On cdb, it'll be possible for modders to add content ratings, but they won't be able to remove ratings. This would stop abuse

**Comment by paramat** [comment]
My request for this is the setting be hardcoded to 'mature' for Android, for the short term at least.

**Comment by SmallJoker** [comment]
How about merging the non-free packages setting into this new filter list? I know that it's not a rating, but this filter system could be extended to more than just content rating.

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -1941,6 +1941,13 @@ num_emerge_threads (Number of emerge threads) int 0
...
+contentdb_url (ContentDB URL) string https://content.minetest.net
+
+#    Comma separated list of flags to hide in the content repository.
+#    "nonfree" can be used to hide packages which do not qualify as 'free software',
 #    as defined by the Free Software Foundation.
-show_nonfree_packages (Show non-free packages) bool false
+#    You can also specify content ratings, which are defined independently
+#    from Minetest versions.
+#    A list can be found at https://content.minetest.net/help/content_flags/
+hide_content_ratings (Show non-free packages) string nonfree, desktop_default
```
Comment (SmallJoker) [inline builtin/settingtypes.txt]: `hide_packages (Hide package flags) string nonfree, desktop_default`
Or possibly a better setting name: `contentdb_flag_blacklist`

**Comment by rubenwardy** [comment]
Implemented Krock's suggestion

Counter-part for non-free filter: https://github.com/minetest/contentdb/commit/9c2c8c21f1e3b412328f05629204a7cf105236c4

**Comment by paramat** [comment]
> I'm still sceptical that adding a user-set filter to CDB allows any content without raising the rating,

Well, i'll trust you on this.
```
Ratings

Content ratings aren't currently supported by ContentDB. Instead, mature content isn't allowed at all for now.

In the future, more mature content will be allowed but labelled with content ratings which may contain the following:

    android_default - meta-rating which includes gore and drugs.
    desktop_default - meta-rating which won't include anything for now.
```
So no objections from me now.

**Pull request merged**